src/v/archival/ntp_archiver_service.cc

```diff
@@ -72,6 +77,67 @@ ntp_archiver::ntp_archiver(
       _start_term);
 }
 
+void ntp_archiver::run_upload_loop() {
+    // NOTE: not using ssx::spawn_with_gate_then here because we want to log
+    // inside the gate (so that _rtclog is guaranteed to be alive).
+    ssx::spawn_with_gate(_gate, [this] {
+        return ss::with_scheduling_group(
+                 _upload_sg, [this] { return upload_loop(); })
+          .handle_exception_type([](const ss::abort_requested_exception&) {})
+          .handle_exception_type([](const ss::sleep_aborted&) {})
+          .handle_exception_type([](const ss::gate_closed_exception&) {})
```

Comment:
Yeah, but because we log errors (on line 92) inside `spawn_with_gate` we have to ignore them once more. And we log inside because we want `_rtclog` to be alive.